lib/sidekiq/throttled/queues_pauser.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require "set"
+require "singleton"
+
+require "sidekiq/throttled/communicator"
+require "sidekiq/throttled/queue_name"
+
+module Sidekiq
+  module Throttled
+    class QueuesPauser
+      include Singleton
+
+      # Redis key of Set with paused queues.
+      # @return [String]
+      PAUSED_QUEUES = "throttled:X:paused_queues".freeze
+      private_constant :PAUSED_QUEUES
+
+      # {Communicator} message used to notify about new queue to be paused.
+      # @return [String]
+      PAUSE_MESSAGE = "pause".freeze
+      private_constant :PAUSE_MESSAGE
+
+      def initialize
+        @paused_queues = Set.new
+      end
+
+      def setup!
+        Sidekiq.configure_server do |config|
+          config.on :startup do
+            @paused_queues.merge paused_queues.map { |q| QueueName.expand q }
```

Comment:
Race condition here as this will (generally) be run prior to the `Communicator` subscription being started (allowing paused queues to be missed if they are added after this but before the subscription starts).